documentation/modules/cruise-control/con-rebalance-performance.adoc

```diff
@@ -0,0 +1,40 @@
+// Module included in the following assemblies:
+//
+// assembly-cruise-control-concepts.adoc
+
+[id='con-rebalance-{context}']
+
+= Rebalance Performance Tuning Overview 
+
+When an xref:con-optimization-proposals-{context}[optimization proposal] is xref:proc-approving-optimization-proposal-{context}[approved], the partition reassignment commands that make up the proposal are applied to the Kafka cluster by the Cruise Control server. These commands consist of two distinct operations:
+
+* Partition movement: This involves physically transferring the partition replica and its data to a new location and can take one of two forms.
+    ** Inter-broker movement: The partition replica is moved to a log directory on another broker.
+    ** Intra-broker movement: The partition replica is moved to a different log directory on the same broker.
+
+* Leadership movement: This involves switching the leader of a set of partition replicas.
+
+Cruise Control will batch up these reassignment commands and issue them to the Kafka cluster. The number of each type of these movements in these batches can effect the performance impact of the rebalance on the Kafka cluster whilst it is enacting them. Cruise control offers several configuration options for tuning the rebalance which can be set at the xref:ref-cruise-control-configuration-{context}[server] or xref:proc-generating-optimization-proposals-{context}[individual rebalance request] level. 
+
+The Cruise Control server setting can be set in the Kafka custom resource under
+`Kafka.spec.kafka.spec.cruiseControl.spec`. The individual rebalance performance configurations can be set under `KafkaRebalance.spec`. The relevant configurations are summarised below:
+
+|==========================================================================================================================
+| Server Configuration                              | KafkaRebalance Configuration            | Description | Default Value 
+
+| `num.concurrent.partition.movements.per.broker`   | `concurrentPartitionMovementsPerBroker` | 
+  The maximum number of inter-broker partition movements in each partition reassignment batch | 10
+
+| `num.concurrent.intra.broker.partition.movements` | `concurrentIntraBrokerPartitionMovements`     | 
+  The maximum number of intra-broker partition movements in each partition reassignment batch | 2
+
+
+| `num.concurrent.leader.movements`                 | `concurrentLeaderMovements`             | 
+  The maximum number of partition leadership changes in each partition reassignment batch     | 1000        
+
+
+| `default.replication.throttle`                    | `replicationThrottle`                   |
+  The bandwidth (in bytes per second) to be assigned to the reassigning of partitions         | No Limit    
+|==========================================================================================================================
```

Comment:
I think that with the latest documentation updates we are avoiding to replicate what we already have in the CRD and the related doc autogenerated in the appendix. Should we just add a link to the appendix here? @laidan6000 @PaulRMellor am I wrong?

Reply:
Can you document a value's default with an annotation in the CRD java class?

Reply:
@tombentley ?

Reply:
I _think_ the DocGenerator will document `@JsonProperty(defaultValue="...")`

Reply:
I added those properties but it didn't seem to feed through to the CRDs (presuming they are recreated during a `make clean all`). What is the build command for (only) forcing a regen of the CRDs?

Reply:
@tomncooper looks like I was wrong. I see no reason why we couldn't change that though, for both the `DocGenerator` and the `CrdGenerator`. It would mean we could make the defaults more explicit e.g. by having a column for them in the tables, rather than them having to be documented in the description. WDYT?

Reply:
I think that would be useful, we have had a few questions about defaults for CC. I will put together a separate PR. I have added the default values to the description for now and they are in the docs.

Reply:
:+1:

Reply:
Noted.